### PR TITLE
Fix some bugs in the code, discovered by just now released new pyflakes

### PR DESCRIPTION
**Pull request by megies**
See e.g. http://tests.obspy.org/36956/#1.
Guess those are from the 14% untested code.. ;-)

### CODE REVIEW

File: obspy/io/cmtsolution/core.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def _is_cmtsolution(filename_or_buf):
                              reset_fp=True)
     # Happens for example when passing the data as a string which would be
     # interpreted as a filename.
-    except (OSError, FileNotFoundError):
+    except OSError:
```
Comment (QuLogic) [inline obspy/io/cmtsolution/core.py]: `FileNotFoundError` is a Python 3 exception and maybe comes from `future`? Though it subclasses from `OSError` anyway.

File: obspy/io/seiscomp/sc3ml.py [hunk 2]
```diff
@@ -890,14 +890,15 @@ def _read_float_var(elem, cls, unit=False, datum=False, additional_mapping={}):
         convert = float(elem)
     except:
         warnings.warn(
-            "'%s' could not be converted to a float. Will be skipped. Please "
-            "contact to report this issue." % etree.tostring(elem),
+            "Encountered a value '%s' which could not be converted to a "
+            "float. Will be skipped. Please contact to report this "
```
Comment (QuLogic) [inline obspy/io/seiscomp/sc3ml.py]: Probably should say _who_ to contact, too...

File: obspy/io/cmtsolution/core.py (hunk 1 above)
Comment (megies) [inline obspy/io/cmtsolution/core.py]: Yeah, makes no difference at all if it's in there or not (as a subclass of `OSError`..)

File: obspy/io/seiscomp/sc3ml.py (hunk 2 above)
Comment (megies) [inline obspy/io/seiscomp/sc3ml.py]: Well, the whole routine should be refactored. Seems like a copy/paste from a routine working with etree elements at some point but now working on string representations of floats. I think @Jollyfant implemented this module, maybe he finds some time for cleaning up this routine at some point, but I think we should do that in a separate issue. This one is really fixing bugs and should not be held up.

**Comment by megies** [comment]
Merging now, the fails on Travis Py3 are already handled in #1285 so we'll get them into maintenance soon, I think.

**Comment by Jollyfant** [comment]
There are definitely some oddities in the sc3ml module. I will try and take a look and work on some improvements soon.

**Pull request merged**